.circleci/soltest_all.sh

```diff
@@ -28,10 +28,46 @@ set -e
 
 REPODIR="$(realpath $(dirname $0)/..)"
 
-EVM=istanbul OPTIMIZE=1 ABI_ENCODER_V2=1 ${REPODIR}/.circleci/soltest.sh
+# This number has to equal the number of steps below.
+STEPS=11
+
+if [ "$CIRCLE_NODE_TOTAL" -a "$CIRCLE_NODE_TOTAL" -gt 1 ]
+then
+    # Run step 1 as the only step on the first executor
+    # and evenly distribute the other steps among
+    # the other executors.
+    # The first step takes much longer than the other steps.
+    if [ "$CIRCLE_NODE_INDEX" = "0" ]
+    then
+        RUN_STEPS="1"
+    else
+        export CIRCLE_NODE_INDEX=$(($CIRCLE_NODE_INDEX - 1))
+        export CIRCLE_NODE_TOTAL=$(($CIRCLE_NODE_TOTAL - 1))
+        RUN_STEPS=$(seq 2 "$STEPS" | circleci tests split)
+    fi
+else
+    RUN_STEPS=$(seq "$STEPS")
+fi
+
+# turn newlines into spaces
+RUN_STEPS=$(echo $RUN_STEPS)
```

Comment:
You could also use `xargs` to get the values on a single line. Maybe it would be a bit more explicit that way:
```suggestion
    RUN_STEPS=$(seq "$STEPS" | xargs)
fi
```

Reply:
I would say echo and xargs are similarly obvious in this situation

Reply:
Maybe `xargs` is not obvious but at least putting stuff on one like is its purpose :)

How about `seq "$STEPS" | tr '\n' ' '` instead? That would be pretty explicit and I think it could go without a comment. At least it's not just a side effect of how shell processes variables.

Reply:
The problem is that the circleci test splitter tool needs newlines, so we need some kind of `echo` here anyway.

Reply:
You could put `tr '\n' '` after `circleci split` rather than after `seq`. But anyway, at this point I'm fine with `echo`. I don't want to drag the discussion out too much :)